src/Features/Core/Portable/ReplaceConditionalWithStatements/AbstractReplaceConditionalWithStatementsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,334 @@
+                }
+            }
+
+            return null;
+        }
+
+        TConditionalExpressionSyntax? CheckNode(SyntaxNode? node)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            if (node is TConditionalExpressionSyntax conditionalExpression)
+                return conditionalExpression;
+
+            if (node is TExpressionSyntax or TArgumentListSyntax or TArgumentSyntax)
+                return Recurse(node);
+
+            return null;
+        }
+    }
+
+    private void TryHandleConditionalExpression(
+        CodeRefactoringContext context,
+        TConditionalExpressionSyntax? conditionalExpression)
+    {
+        if (conditionalExpression is null)
+            return;
+
+        var document = context.Document;
+        var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+        // Walk upwards from this conditional, through parent expressions and arguments until we find the owning construct.
+        var topExpression = GetTopExpression(conditionalExpression);
+
+        // Check if we're parented by one of the simple statement types.  In all these cases, we just convert the
+        //
+        //      ... a ? b : c ...;
+        //
+        // into
+        //
+        //      if (a)
+        //          ... b ...;
+        //      else
+        //          ... c ...;
+        //
+        // (in other words, we duplicate the statement wholesale, just replacing the conditional with the
+        // when-true/false portions.
+        var parentStatement = topExpression.Parent as TStatementSyntax;
+        if (IsSupportedSimpleStatement(syntaxFacts, parentStatement))
+        {
+            context.RegisterRefactoring(CodeAction.Create(
+                FeaturesResources.Replace_conditional_expression_with_statements,
+                c => ReplaceConditionalExpressionInSingleStatementAsync(document, conditionalExpression, parentStatement, c)),
+                conditionalExpression.Span);
+            return;
+        }
+
+        // However, if we're parented by a local decl, e.g.:
+        //
+        //      object v = a ? b : c;
+        //
+        // Then we want to break this into two statements.  One for the declaration, and one for the if-statement:
+        //
+        //      object v;
+        //      if (a)
+        //          v = b;
+        //      else
+        //          v = c;
+
+        if (topExpression.Parent is TEqualsValueClauseSyntax equalsValue &&
+            equalsValue.Parent is TVariableDeclaratorSyntax variableDeclarator &&
+            conditionalExpression.GetAncestor<TLocalDeclarationStatementSyntax>() is { } localDeclarationStatement &&
+            HasSingleVariable(localDeclarationStatement, out var variable) &&
+            syntaxFacts.IsDeclaratorOfLocalDeclarationStatement(variableDeclarator, localDeclarationStatement) &&
+            CanRewriteLocalDeclarationStatement(localDeclarationStatement))
```

Comment:
Not sure how to simplify this since it's so nested, but it took a little bit to wrap my head around.